### PR TITLE
jigsawstack[major]: Add Jigsawtack Integration

### PR DESCRIPTION
**Pull request by iamNarcisse**
## Add JigsawStack  agent tools. 

Leveraging  [JigsawStack](https://jigsawstack.com/) AI SDK,  this PR adds the following agent tools:

1. **JigsawStackAIScrape** - This tool enables you to scrape web content using advanced AI capabilities provided by JigsawStack.

2. **JigsawStackAISearch** - This tool allows you to perform web searches and retrieve high-quality results powered by AI.

3. **JigsawStackSpeechToText** - Transcribe video and audio files using the Whisper large V3 AI model.

4. **JigsawStackVOCR** - This tool accurately recognizes, describes, and retrieves data from images.

5. **JigsawStackTextToSQL** - This tool helps generate semantically correct SQL queries from text.

6. **JigsawStackPromptEngine** - This tool generates text completions with support for structured outputs.

### CODE REVIEW

File: deno.lock
Comment (jacoblee93) [inline deno.lock:1]: Can you remove this?

File: libs/langchain-jigsawstack/src/llms.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  implements JigsawStackPromptEngineInput
+{
+  static lc_name() {
+    return "JigsawStackPromptEngine";
+  }
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {
+      apiKey: "JIGSAWSTACK_API_KEY",
+      api_key: "JIGSAWSTACK_API_KEY",
```
Comment (jacoblee93) [inline libs/langchain-jigsawstack/src/llms.ts:45]: You don't need the snake cased version of this

File: libs/langchain-jigsawstack/src/llms.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    };
+  }
+
+  get lc_aliases(): { [key: string]: string } | undefined {
```
Comment (jacoblee93) [inline libs/langchain-jigsawstack/src/llms.ts:49]: You don't need this

**Review by jacoblee93 — CHANGES_REQUESTED** [review]
Thanks for the PR!

We'd need to write docs for this - but we've also been trying to move new partner integrations into separate repos to make management a bit easier. You could still add docs to the main `langchain` repo and they'd appear under:

https://js.langchain.com/v0.2/docs/integrations/platforms/

You could then manage your own releases as `@jigsawstack/langchain` or something similar, or could grant us access to the repo and we could push releases as `@langchain/jigsawstack`, whichever you prefer.

File: deno.lock
Comment (iamNarcisse) [reply deno.lock:1]: Sure will do

File: libs/langchain-jigsawstack/src/llms.ts (hunk 2 above)
Comment (iamNarcisse) [reply libs/langchain-jigsawstack/src/llms.ts:49]: Alright.

**Comment by iamNarcisse** [comment]
> Thanks for the PR!
> 
> We'd need to write docs for this - but we've also been trying to move new partner integrations into separate repos to make management a bit easier. You could still add docs to the main `langchain` repo and they'd appear under:
> 
> https://js.langchain.com/v0.2/docs/integrations/platforms/
> 
> You could then manage your own releases as `@jigsawstack/langchain` or something similar, or could grant us access to the repo and we could push releases as `@langchain/jigsawstack`, whichever you prefer.
> 
> We could also move these into `@langchain/community`.

Hi @jacoblee93, 

I have made the requested changes. Let me know if it's all good. 

On how releases would be deployed, We'd prefer releases  as `@langchain/jigsawstack`

**Comment by jacoblee93** [comment]
> On how releases would be deployed, We'd prefer releases as @langchain/jigsawstack. The docs have been prepared and will be included in a separate PR.

In that case, could you make a separate repo and add add me as an admin? I'll be able to cut releases from there.

**Comment by iamNarcisse** [comment]
> > On how releases would be deployed, We'd prefer releases as @langchain/jigsawstack. The docs have been prepared and will be included in a separate PR.
> 
> In that case, could you make a separate repo and add add me as an admin? I'll be able to cut releases from there.

Alright

**Comment by iamNarcisse** [comment]
@jacoblee93, made PR for the docs here  #6767

**Review by bracesproul — CHANGES_REQUESTED** [review]
@iamNarcisse if you want the code to be released as `@langchain/jigsawstack` you'll have to remove the `libs/langchain-jigsawstack` dir in this PR & create a separate repo (managed by you, but with admin perms granted to jacob/myself. Alternatively you can move the code into `libs/langchain-community`.

Please pig me when one of these has been done & I can take a look!

**Comment by iamNarcisse** [comment]
> @iamNarcisse if you want the code to be released as `@langchain/jigsawstack` you'll have to remove the `libs/langchain-jigsawstack` dir in this PR & create a separate repo (managed by you, but with admin perms granted to jacob/myself. Alternatively you can move the code into `libs/langchain-community`.
> 
> Please pig me when one of these has been done & I can take a look!

Hi @bracesproul, thanks for the feedback. I have removed dir `libs/langchain-jigsawstack`  and made a separate PR for the docs, see #6767. I have added @jacoblee93 to the forked repo that contains the code.

**Pull request closed without merging**